sklearn/linear_model/tests/test_perceptron.py

```diff
@@ -67,3 +67,13 @@ def test_undefined_methods():
     clf = Perceptron(max_iter=100)
     for meth in ("predict_proba", "predict_log_proba"):
         assert_raises(AttributeError, lambda x: getattr(clf, x), meth)
+
+
+def test_perceptron_l1_ratio():
+    clf1 = Perceptron(l1_ratio=0, penalty='elasticnet')
+    clf1.fit(X, y)
+
+    clf2 = Perceptron(l1_ratio=0.15, penalty='elasticnet')
+    clf2.fit(X, y)
+
+    assert clf1.score(X, y) != clf2.score(X, y)
```

Comment:
It would be better to assert that the scores are different with a significant difference (e.g. something like .1 or .05).

`     assert clf1.score(X, y) != clf2.score(X, y) ` would pass even if the score difference was 1e-15 and that's not  strict enough

Reply:
`clf1.score(X, y)` is 0.96 and `clf2.score(X, y)` is 0.906.
How about `assert clf1.score(X, y) - clf2.score(X, y) > 0.01` ?

Reply:
that sounds fine. Maybe check the absolute difference?

Reply:
`clf1` is overfitted more than `clf2`. Is it normal that `clf2.score(X, y)` is greater than `clf1.score(X, y)`?

Reply:
Would it make sense to have `l1_ratio=0` and check with `penalty="l2"` and `l1_ratio=1` and check with `penalty="l1"`?

Reply:
Hi @glemaitre , 
Do you mean something like 
`assert Perceptron(l1_ratio=0, penalty='elasticnet').fit(X, y).score(X, y) != Perceptron(l1_ratio=0, penalty='l2').fit(X, y).score(X, y)` 
and 
`assert Perceptron(l1_ratio=1, penalty='elasticnet').fit(X, y).score(X, y) != Perceptron(l1_ratio=1, penalty='l1').fit(X, y).score(X, y)`?

Reply:
I pushed a small test to check that this is passing across all platform.
We should be able to check the coefficient in l1 and with l1_ratio=1 and l2 with l1_ratio=0